client/tests/announcements.spec.ts

```diff
@@ -0,0 +1,96 @@
+import mockRequests from "./helpers/mocks";
+import test from "./helpers/test";
+import { expect } from "@playwright/test";
+
+test.describe("Announcements", () => {
+  test("Should display announcements page with table", async ({ page, helpers }) => {
+    await mockRequests(page, {
+      announcements: [
+        {
+          id: 1,
+          title: "Welcome to Food Oasis!",
+          description: "This is a test announcement.",
+          created_at: "2025-07-19T10:30:00.000Z",
+          severity: "info",
+          is_enabled: true,
+        },
+        {
+          id: 2,
+          title: "System Update",
+          description: "We have updated our system.",
+          created_at: "2025-07-18T15:45:00.000Z",
+          severity: "warning",
+          is_enabled: true,
+        },
+      ],
+    });
+
+    await helpers.mockLogin();
+    await page.goto("/admin/announcements");
```

Comment:
I just realized the login state is not persistent in the tests. We need to fix that 🫣 but for now, you can do this instead of navigating to another page 👌 
```
    await page.getByTestId("menu-button").click();
    await page.getByRole("link", { name: "Announcements" }).click();
```